### ISSUE TITLE
`if` condition should be `i32`

### ISSUE THREAD

**Issue by binji** [WebAssembly/binaryen#497]
This test doesn't fail:
```
(module
  (func
    (if (f32.const 0) (i32.const 0) (i32.const 0))))
```

**Comment by sunfishcode** [comment]
Do we have spec tests covering this?

**Comment by binji** [comment]
No, I guess I should probably write up some spec tests for all these.

**Comment by kripken** [comment]
Ok, these all should be fixed by #499.

Our validation was indeed pretty poor before, it was basically just enough to pass the spec test suite ;) thanks!

**Comment by binji** [comment]
Some new spec tests: https://github.com/WebAssembly/spec/pull/289

**Comment by kripken** [comment]
Fixed by #499.
